i2c/oled_i2c/README.adoc

```diff
@@ -0,0 +1,68 @@
+    COM 0 |      |      |      |      |
+    COM 1 |      |      |      |      |
+       :  |      |      |      |      |
+    COM 7 |      |      |      |      |
+    ------------------------------------
+----
+
+Note the difference between columns in RAM and the actual segment pins on the display. The distinction between these two is important as we can remap, or add offsets between the two in the driver itself instead of moving pixels in our frame buffer code.
+
+The driver has 3 modes of transferring the pixels in RAM to the display on clock cycle (provided that the driver is set to use its RAM content to drive the display, ie. command 0xA4 is sent). We choose horizontal addressing mode which, after setting the column address and page address registers to our desired start positions, will increment the column address register until the OLED display width is reached (127 in our case) after which the column address register will reset to its starting value and the page address is incremented. Once the page register reaches the end, it will wrap around as well. Effectively, this scans across the display from top to bottom, left to right in blocks that are 8 pixels high. When a byte is sent to be written into RAM, it sets all the rows for the current position of the column address register. So, if we send 10101010, and we are on PAGE 0 and COL1, COM0 is set to 1, COM1 is set to 0, COM2 is set to 1, and so on. Effectively, the byte is "transposed" to fill a single page's column. The datasheet has further information on this and the two other modes.
+
+Horizontal addressing mode has the key advantage that we can keep one single 512 byte buffer (128 columns x 4 pages and each byte fills a page's rows) and write this in one go to the RAM (column address auto increments on writes as well as reads) instead of working with 2D matrices of pixels and adding more overhead. 
+
+[NOTE]
+======
+* The SSD1306 is able to drive 128x64 displays but as our display is 128x32, only half of the COM (common) pins are connected to the display.
+* The specific display model being used is UG-2832HSWEG02
+======
+
+== Wiring information
+
+Wiring up the device requires 4 jumpers, to connect VCC (3.3v), GND, SDA and SCL and optionally a 5th jumper for the driver RESET pin. The example here uses the default I2C port 0, which is assigned to GPIO 4 (SDA) and 5 (SCL) in software. Power is supplied from the 3.3V pin from the Pico.
+
+[[oled_i2c_wiring]]
+[pdfwidth=75%]
+.Wiring Diagram for oled display via I2C.
+image::oled_i2c_bb.png[]
+
+== List of Files
+
+CMakeLists.txt:: CMake file to incorporate the example into the examples build tree.
+oled_i2c.c:: The example code.
+
+== Bill of Materials
+
+.A list of materials required for the example
+[[oled_i2c-bom-table]]
+[cols=3]
+|===
+| *Item* | *Quantity* | Details
+| Breadboard | 1 | generic part
+| Raspberry Pi Pico | 1 | http://raspberrypi.org/
+| OLED display board | 1 | generic part
```

Comment:
We're using the Adafruit part, and their Fritzing part in the wiring diagram so we should link to the Adafruit part in the BoM. Which I think is this one? https://www.adafruit.com/product/4440

Reply:
Also, I suspect "OLED display board" is a **much** too generic description, as there are probably many many different OLED display boards? :laughing:

Reply:
Many! That it's a SSD1306-driven board is probably more of a differentiator. Hence I think we should link to the specific part in the BoM, with discussion that the example code will probably work for other SSD1306-based displays in the main text.

Reply:
Makes sense. I wasn't sure what the policy was on mentioning third parties in the docs so I left it deliberately ambiguous. Fixed now.